File: src/cpp_solutions/1_two-sum.cpp
#include <bits/stdc++.h>

using namespace std;

// start_submission
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        map<int, int> m;

        for (int i = 0; i < nums.size(); ++i) {
            int x = nums[i];

            int comp = target - x;
            if (m.find(comp) != m.end()) {
                return {i, m[comp]};
            }

            m[x] = i;
        }

        __builtin_unreachable();
    }
};
// end_submission

File: src/rust_solutions/1_two-sum.rs
#[allow(unused)]
use crate::stubs::*;

struct Solution;

// start_submission
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut m: HashMap<i32, i32> = HashMap::new();
        for (idx, &num) in nums.iter().enumerate() {
            let comp: i32 = target - num;
            if m.contains_key(&comp) {
                return vec![*m.get(&comp).unwrap(), idx as i32];
            }
            m.insert(num, idx as i32);
        }

        vec![]
    }
}
// end_submission
